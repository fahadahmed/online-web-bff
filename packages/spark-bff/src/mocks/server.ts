import { setupServer } from 'msw/node';
import a2pCustomerHandlers from 'datasources/a2p/customer/mocks/handlers';
import accessHandlers from 'datasources/user/access/mocks/handlers';
import accountSummaryHandlers from 'datasources/customer/accountsService/mocks/handlers';
import addressHandlers from 'datasources/location/address/mocks/handlers';
import airpointsHandlers from 'datasources/airpoints/mocks/handlers';
import availableNumbers from 'datasources/shop/availableNumbers/mocks/handlers';
import balanceServiceHandlers from 'datasources/finance/balanceService/mocks/handlers';
import billPaymentHandlers from 'datasources/finance/paymentService/billPayment/mocks/handlers';
import cartServicesHandlers from 'datasources/shop/cart/mocks/handlers';
import checkoutServicesHandlers from 'datasources/shop/checkout/mocks/handlers';
import connectionDiagnosticsHandlers from 'datasources/line/assure/connectionDiagnostics/mocks/handlers';
import connectionPromiseHandler from 'datasources/line/assure/connectionPromise/mocks/handlers';
import contentServiceHandlers from 'datasources/utility/content/mocks/handlers';
import digitalBillHandlers from 'datasources/finance/digitalBill/mocks/handlers';
import identityServiceHandlers from 'datasources/user/identityService/mocks/handlers';
import lineConfiguarationHandlers from 'datasources/line/lineConfiguration/mocks/handlers';
import lineDetailServiceHandlers from 'datasources/line/lineDetails/mocks/handlers';
import lineDetailsHandlers from 'datasources/line/lineSummary/mocks/handlers';
import lineSpendHistoryHandlers from 'datasources/line/spend/mocks/handlers';
import lineSubscriptionsService from 'datasources/line/lineSubscriptionsService/mocks/handlers';
import lineUsageDetailsMocks from 'datasources/line/usage/usageDetailsService/mocks';
import lineUsageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';
import lineUsageMocks from 'datasources/line/usage/mocks/handlers';
import lineUsageTransactionMocks from 'datasources/line/usage/usageTransactionsService/mocks';
import networkSettingsHandlers from 'datasources/technology/networkSettingsService/mocks/handlers';
import orderHandlers from 'datasources/shop/order/mocks/handlers';
import paymentServiceHandlers from 'datasources/finance/paymentService/mocks/handlers';
import pointOfAddressHandlers from 'datasources/location/pointOfInterest/mocks/handlers';
import pointOfOutageHandlers from 'datasources/location/pointOfInterest/outagesPointOfInterest/mocks/handlers';
import preferenceHandlers from 'datasources/user/preference/mocks/handlers';
import prepaidBalanceServiceHandlers from 'datasources/finance/prepaidBalanceService/mocks/handlers';
import productDetailServiceHandlers from 'datasources/product/details/mocks/handlers';
import productFiltersServiceHandlers from 'datasources/product/filters/mocks/handlers';
import productOffersHandlers from 'datasources/product/offers/mocks/handlers';
import productPromotionsMPDServiceHandlers from 'datasources/product/promotions/mpd/mocks/handlers';
import profileServiceHandlers from 'datasources/marketing/profileService/mocks/handlers';
import resourceHandlers from 'datasources/utility/resource/mocks/handlers';
import roamingRatesHandlers from 'datasources/product/roamingRates/mocks/handlers';
import subscriptionActivationHandlers from 'datasources/line/subscriptionActivationService/mocks/handlers';
import topupHandlers from 'datasources/finance/paymentService/topup/mocks/handlers';
import transactionsListHandlers from 'datasources/line/transactionsService/mocks/handlers';
import troubleshootHandlers from 'datasources/line/assure/connectionTroubleshooter/mocks/handlers';
import userInfoHandlers from 'datasources/user/userInfo/mocks/handlers';
import walletServiceHandlers from 'datasources/finance/walletService/mocks/handlers';

const handlers = [
  ...a2pCustomerHandlers,
  ...accessHandlers,
  ...accountSummaryHandlers,
  ...addressHandlers,
  ...availableNumbers,
  ...airpointsHandlers,
  ...balanceServiceHandlers,
  ...billPaymentHandlers,
  ...cartServicesHandlers,
  ...checkoutServicesHandlers,
  ...connectionDiagnosticsHandlers,
  ...connectionPromiseHandler,
  ...contentServiceHandlers,
  ...digitalBillHandlers,
  ...identityServiceHandlers,
  ...lineConfiguarationHandlers,
  ...lineDetailServiceHandlers,
  ...lineDetailsHandlers,
  ...lineSpendHistoryHandlers,
  ...lineSubscriptionsService,
  ...lineUsageDetailsMocks,
  ...lineUsageHistoryMocks,
  ...lineUsageMocks,
  ...lineUsageTransactionMocks,
  ...networkSettingsHandlers,
  ...orderHandlers,
  ...paymentServiceHandlers,
  ...pointOfAddressHandlers,
  ...pointOfOutageHandlers,
  ...preferenceHandlers,
  ...prepaidBalanceServiceHandlers,
  ...productDetailServiceHandlers,
  ...productFiltersServiceHandlers,
  ...productOffersHandlers,
  ...productPromotionsMPDServiceHandlers,
  ...profileServiceHandlers,
  ...resourceHandlers,
  ...roamingRatesHandlers,
  ...subscriptionActivationHandlers,
  ...topupHandlers,
  ...transactionsListHandlers,
  ...troubleshootHandlers,
  ...userInfoHandlers,
  ...walletServiceHandlers,
];

const server = setupServer(...handlers);
export default server;
