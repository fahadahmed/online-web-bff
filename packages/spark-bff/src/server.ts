import { ApolloServer } from 'apollo-server-fastify';
import {
  ApolloServerPluginLandingPageDisabled,
  ApolloServerPluginLandingPageGraphQLPlayground,
} from 'apollo-server-core';
import { logger } from 'apolloLogger';
import context from 'context';
import dataSources from 'datasources';
import { ApolloLoggerPlugin, ApplicationLogEvent } from 'logging';
import schema from 'schema';
import { formatResponse } from 'utils/formatResponse';
import { isDevelopmentEnvironment } from 'utils/isDevelopmentEnvironment';

process.on('uncaughtException', (err) => {
  logger.error(new ApplicationLogEvent({ message: err.message, context: {} }));
  throw err;
});

const apolloServer = new ApolloServer({
  schema,
  dataSources,
  formatResponse,
  logger,
  context,
  introspection: isDevelopmentEnvironment(),
  plugins: [
    new ApolloLoggerPlugin(
      `BFF Server Started. BUILD_ID: ${process.env.BUILD_BUILDID}`,
    ),
    isDevelopmentEnvironment()
      ? ApolloServerPluginLandingPageGraphQLPlayground()
      : ApolloServerPluginLandingPageDisabled(),
  ],
});

export default apolloServer;
