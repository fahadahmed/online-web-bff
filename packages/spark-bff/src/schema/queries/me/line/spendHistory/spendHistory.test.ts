import { server } from 'test/server';
import spendHistoryMocks from 'datasources/line/spend/mocks/handlers';
import { createTestClient, gql } from 'test/apolloTestUtils';

beforeEach(() => {
  server.use(...spendHistoryMocks);
});

test('returns usage charge history for a line ', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "0277489070") {
            spendHistory(
              interval: MONTHLY
              start: "2020-03-01T00:00:00.000+13:00"
              end: "2020-03-01T00:00:00.000+13:00"
            ) {
              interval
              accountType
              gstInclusive
              startDateTime
              endDateTime
              averagePeriodSpend
              summarisedPeriods {
                startDateTime
                endDateTime
                periodSpend
                unbilled
                periodBreakdown {
                  breakdownType
                  periodBreakdownId
                  periodSpend
                }
              }
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});

test('returns usage charge history detail for a line ', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "0277489070") {
            spendHistoryDetail(
              interval: MONTHLY
              periodBreakdownId: "fgNaGUKJHUWCj5WhoE1lZw"
            ) {
              interval
              breakdownType
              gstInclusive
              startDateTime
              endDateTime
              periodSpend
              unbilled
              contributingProducts {
                offerId
                productName
                chargeStartDateTime
                chargeEndDateTime
                productEndDateTime
                periodSpend
              }
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});
