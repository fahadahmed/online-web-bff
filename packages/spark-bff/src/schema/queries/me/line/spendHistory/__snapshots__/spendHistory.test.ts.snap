// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns usage charge history detail for a line  1`] = `
Object {
  "spendHistoryDetail": Object {
    "breakdownType": "PLAN",
    "contributingProducts": Array [
      Object {
        "chargeEndDateTime": "2021-11-06T11:00:00Z",
        "chargeStartDateTime": "2021-10-07T11:00:00Z",
        "offerId": "mbundle050820",
        "periodSpend": 59.99,
        "productEndDateTime": null,
        "productName": "Endless Mobile Plan",
      },
    ],
    "endDateTime": "2021-10-08T10:59:59Z",
    "gstInclusive": true,
    "interval": "MONTHLY",
    "periodSpend": 59.99,
    "startDateTime": "2021-09-08T12:00:00Z",
    "unbilled": false,
  },
}
`;

exports[`returns usage charge history for a line  1`] = `
Object {
  "spendHistory": Object {
    "accountType": "POSTPAID",
    "averagePeriodSpend": 10,
    "endDateTime": "2021-10-31T23:59:59+13:00",
    "gstInclusive": true,
    "interval": "MONTHLY",
    "startDateTime": "2020-10-01T00:00:00+13:00",
    "summarisedPeriods": Array [
      Object {
        "endDateTime": "2021-11-08T10:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2021-10-08T11:00:00Z",
        "unbilled": true,
      },
      Object {
        "endDateTime": "2021-10-08T10:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "ADDITIONAL_PAID",
            "periodBreakdownId": "ADDITIONAL_PAID|2021-09-08T12:00:00.000Z|2021-10-08T10:59:59.000Z",
            "periodSpend": 10,
          },
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-09-08T12:00:00.000Z|2021-10-08T10:59:59.000Z",
            "periodSpend": 59.99,
          },
        ],
        "periodSpend": 69.99,
        "startDateTime": "2021-09-08T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-09-08T11:59:59Z",
        "periodBreakdown": Array [
          Object {
            "breakdownType": "PLAN",
            "periodBreakdownId": "PLAN|2021-08-08T12:00:00.000Z|2021-09-08T11:59:59.000Z",
            "periodSpend": 59.99,
          },
        ],
        "periodSpend": 59.99,
        "startDateTime": "2021-08-08T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-08-08T11:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2021-07-08T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-07-08T11:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2021-06-08T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-06-08T11:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2021-05-08T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-05-08T11:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2021-04-08T12:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-04-08T11:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2021-03-08T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-03-08T10:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2021-02-08T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-02-08T10:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2021-01-08T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2021-01-08T10:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2020-12-08T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2020-12-08T10:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2020-11-08T11:00:00Z",
        "unbilled": false,
      },
      Object {
        "endDateTime": "2020-11-08T10:59:59Z",
        "periodBreakdown": Array [],
        "periodSpend": 0,
        "startDateTime": "2020-10-08T11:00:00Z",
        "unbilled": false,
      },
    ],
  },
}
`;
